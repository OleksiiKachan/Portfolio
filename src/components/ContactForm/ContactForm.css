.contactFormContainer {
    width: 100%;
    margin: 0 auto;
    position: relative;
}

.contactForm {
    padding: 25px;
}

.contactForm__title {
    display: block;
    margin-bottom: 10px;
}

.contactForm__contactField {
    border: medium none !important;
    margin: 0 0 10px;
    min-width: 100%;
    padding: 0;
    width: 100%;
}

.contactForm__contactField_button {
	width: calc(100% + 20px);
}

.contactForm__contactControl {
    width: 100%;
    border: 1px solid #ccc;
    background-color: #FFF;
    margin: 0 0 5px;
    padding: 10px;
}

.contactForm__contactControl:hover {
    transition: border-color 0.3s ease-in-out;
    border: 1px solid #aaa;
}

.contactForm__contactControl_multipleRow {
    height: 100px;
    max-width: 100%;
    resize: none;
}

.contactForm__submitButton {
    cursor: pointer;
    border: none;
    background-color: #4CAF50;
    color: #FFF;
    font-size: 15px;
}

.contactForm__submitButton:hover {
    background-color: #43A047;
    transition: background-color 0.3s ease-in-out;
}

.contactForm__submitButton:active {
    box-shadow: inset 0 1px 3px rgba(0, 0, 0, 0.5);
}

.contactForm__contactControl:focus {
    outline: 0;
    border: 1px solid #aaa;
}

::-webkit-input-placeholder {
    color: #888;
}

:-moz-placeholder {
    color: #888;
}

::-moz-placeholder {
    color: #888;
}

:-ms-input-placeholder {
    color: #888;
}