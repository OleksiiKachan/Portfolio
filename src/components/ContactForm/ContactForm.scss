.contactFormContainer {
  background: rgba(66, 66, 66, 0.8);
  box-sizing: border-box;
  box-shadow: 0 15px 25px rgba(0, 0, 0, 0.5);
  padding: 25px 40px;
}

.contactFormContainer__title {
  margin: 0 0 30px;
  padding: 0;
  color: #ddd;
}

.contactForm {
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  align-items: stretch;
  height: 85%;
}

.contactForm__field {
  position: relative;
}

.contactForm__input {
  width: 100%;
  padding: 10px 0;
  font-size: 16px;
  color: #ddd;
  margin-bottom: 30px;
  border: none;
  border-bottom: 1px solid #ddd;
  outline: none;
  background: transparent;
  letter-spacing: 1px;
}

.contactForm__label {
  position: absolute;
  top: 0;
  left: 0;
  padding: 10px 0;
  font-size: 14px;
  color: #ddd;
  pointer-events: none;
  transition: 0.5s;
  letter-spacing: 1px;
}

.contactForm__input:focus ~ .contactForm__label,
.contactForm__input:valid ~ .contactForm__label {
  top: -18px;
  left: 0;
  color: #03a9f4;
  font-size: 10px;
}

.contactForm__submitButton {
  background: #d5d288;
  border: none;
  outline: none;
  color: #424242;
  font-weight: bolder;
  padding: 14px 20px;
  cursor: pointer;
  box-shadow: 0 4px 4px rgba(0, 0, 0, 0.25);
}
