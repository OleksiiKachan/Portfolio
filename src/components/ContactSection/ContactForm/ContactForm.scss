@import '../../../assets/style/variables';

.contactForm {
  background: rgba(66, 66, 66, 0.8);
  box-sizing: border-box;
  box-shadow: 0 15px 25px rgba(0, 0, 0, 0.5);
  padding: 25px 40px;

  &__title {
    margin: 0 0 30px;
    padding: 0;
    color: #ddd;
    font-size: 1.5rem;
  }

  &__form {
    display: flex;
    flex-direction: column;
    justify-content: space-around;
    align-items: stretch;
    height: 85%;
  }
}

@media screen and (max-width: 768px) {
  .contactForm {
    &__title {
      font-size: 1rem;
    }
  }
}
