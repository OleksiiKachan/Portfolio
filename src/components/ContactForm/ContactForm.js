import React, { Component } from 'react';
import classNames from 'classnames';
import './ContactForm.css';

/*
    <ContactForm />
*/
class ContactForm extends Component {
    render() {
        return (
            <div className={classNames('contactFormContainer')}>
                <form id='contactForm' className={classNames('contactForm')}  action='https://formspree.io/alexkachan.lutsk@gmail.com' method='post'>
                    <h3 className={classNames('contactForm__title')}>Contact me</h3>
                    <fieldset className={classNames('contactForm__contactField')}>
                        <input className={classNames('contactForm__contactControl')} placeholder="Your name" type="text" tabindex="1" name='name' required autofocus/>
                    </fieldset>
                    <fieldset className={classNames('contactForm__contactField')}>
                        <input className={classNames('contactForm__contactControl')} placeholder="Your Email Address" type="email" tabindex="2" name='email' required/>
                    </fieldset>
                    <fieldset className={classNames('contactForm__contactField')}>
                        <input className={classNames('contactForm__contactControl')} placeholder="Your Phone Number (optional)" type="tel" tabindex="3" name='phone'/>
                    </fieldset>
                    <fieldset className={classNames('contactForm__contactField')}>
                        <input className={classNames('contactForm__contactControl')} placeholder="Your Web Site (optional)" type="url" tabindex="4" name='site'/>
                    </fieldset>
                    <fieldset className={classNames('contactForm__contactField')}>
                        <textarea className={classNames('contactForm__contactControl', 'contactForm__contactControl_multipleRow')}  placeholder="Type your message here...." tabindex="5" name='message'></textarea>
                    </fieldset>
                    <fieldset className={classNames('contactForm__contactField', 'contactForm__contactField_button')}>
                        <button className={classNames('contactForm__contactControl', 'contactForm__submitButton')} type="submit" id="contact-submit" data-submit="...Sending">Submit</button>
                    </fieldset>
                </form>
            </div>
        );
    }
}

export default ContactForm;