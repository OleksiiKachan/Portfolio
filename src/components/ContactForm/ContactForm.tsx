import React, { Component } from 'react';
import classNames from 'classnames';
import './ContactForm.scss';

/*
    <ContactForm />
*/
class ContactForm extends Component<any, any> {
  render() {
    return (
      <div className={classNames('contactFormContainer')}>
        <form
          id="contactForm"
          className={classNames('contactForm')}
          action="https://formspree.io/alexkachan.lutsk@gmail.com"
          method="post"
        >
          <h3 className={classNames('contactForm__title')}>Contact me</h3>
          <fieldset className={classNames('contactForm__contactField')}>
            <input
              className={classNames('contactForm__contactControl')}
              placeholder="Your name"
              type="text"
              name="name"
              required
            />
          </fieldset>
          <fieldset className={classNames('contactForm__contactField')}>
            <input
              className={classNames('contactForm__contactControl')}
              placeholder="Your Email Address"
              type="email"
              name="email"
              required
            />
          </fieldset>
          <fieldset className={classNames('contactForm__contactField')}>
            <input
              className={classNames('contactForm__contactControl')}
              placeholder="Your Phone Number (optional)"
              type="tel"
              name="phone"
            />
          </fieldset>
          <fieldset className={classNames('contactForm__contactField')}>
            <input
              className={classNames('contactForm__contactControl')}
              placeholder="Your Web Site (optional)"
              type="url"
              name="site"
            />
          </fieldset>
          <fieldset className={classNames('contactForm__contactField')}>
            <textarea
              className={classNames(
                'contactForm__contactControl',
                'contactForm__contactControl_multipleRow'
              )}
              placeholder="Type your message here...."
              name="message"
            />
          </fieldset>
          <fieldset
            className={classNames(
              'contactForm__contactField',
              'contactForm__contactField_button'
            )}
          >
            <button
              className={classNames(
                'contactForm__contactControl',
                'contactForm__submitButton'
              )}
              type="submit"
              id="contact-submit"
              data-submit="...Sending"
            >
              Submit
            </button>
          </fieldset>
        </form>
      </div>
    );
  }
}

export default ContactForm;
