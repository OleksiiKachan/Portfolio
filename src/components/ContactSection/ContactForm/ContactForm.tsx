import React, { Component } from 'react';
import classNames from 'classnames';
import OutlineButton from '../../OutlineButton';
import Textbox from '../../Textbox';
import './ContactForm.scss';

/*
    <ContactForm
      className=''
    />
*/

type PropsType = {
  className?: string;
};

export default ({ className }: PropsType) => {
  return (
    <div className={classNames('contactForm', className)}>
      <h2 className={classNames('contactForm__title')}>Contact Me</h2>
      <form
        className={classNames('contactForm__form')}
        autoComplete="off"
        action="https://formspree.io/alexkachan.lutsk@gmail.com"
        method="post"
      >
        <div>
          <Textbox fieldName="name" label="Your name" required />
          <Textbox fieldName="email" label="Your email" type="email" required />
        </div>
        <Textbox fieldName="message" label="Your message" required />
        <OutlineButton type="gold" onClick={() => {}} isSubmit>
          Submit
        </OutlineButton>
      </form>
    </div>
  );
};
