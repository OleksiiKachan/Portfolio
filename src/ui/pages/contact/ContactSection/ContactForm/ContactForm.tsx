import React from 'react';
import classNames from 'classnames';
import { OutlineButton, Textbox } from '../../../../elements';
import './ContactForm.scss';

/*
    <ContactForm
      className=''
    />
*/

type PropsType = {
  className?: string;
};

export default ({ className }: PropsType) => {
  return (
    <div className={classNames('contactForm', className)}>
      <h3 className={classNames('contactForm__title')}>Contact Me</h3>
      <form
        className={classNames('contactForm__form')}
        autoComplete="off"
        action="https://formspree.io/alexkachan.lutsk@gmail.com"
        method="POST"
      >
        <div>
          <Textbox fieldName="name" label="Your name" required />
          <Textbox
            fieldName="_replyto"
            label="Your email"
            type="email"
            required
          />
        </div>
        <Textbox fieldName="message" label="Your message" required />
        <OutlineButton type="gold" onClick={() => {}} isSubmit>
          Submit
        </OutlineButton>
      </form>
    </div>
  );
};
