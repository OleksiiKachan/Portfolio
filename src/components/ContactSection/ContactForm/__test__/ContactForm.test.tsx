import React from 'react';
import { shallow } from 'enzyme';
import ContactForm from '..';

describe('ContactForm', () => {
  it('Snapshot', () => {
    const component = shallow(<ContactForm />);

    expect(component).toMatchSnapshot();
  });
});
