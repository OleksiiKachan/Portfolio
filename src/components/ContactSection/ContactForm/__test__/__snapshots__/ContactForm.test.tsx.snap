// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactForm Snapshot 1`] = `
<div
  className="contactForm"
>
  <h2
    className="contactForm__title"
  >
    Contact Me
  </h2>
  <form
    action="https://formspree.io/alexkachan.lutsk@gmail.com"
    autoComplete="off"
    className="contactForm__form"
    method="post"
  >
    <div>
      <_default
        fieldName="name"
        label="Your name"
        required={true}
      />
      <_default
        fieldName="email"
        label="Your email"
        required={true}
        type="email"
      />
    </div>
    <_default
      fieldName="message"
      label="Your message"
      required={true}
    />
    <_default
      isSubmit={true}
      onClick={[Function]}
      type="gold"
    >
      Submit
    </_default>
  </form>
</div>
`;
